src/test/java/io/vertx/ext/couchdb/CouchdbClientIT.java

```diff
@@ -0,0 +1,139 @@
+package io.vertx.ext.couchdb;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.couchdb.utils.IntegrationTest;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@IntegrationTest
+@Testcontainers
+public class CouchdbClientIT {
+
+  @Container
+  private static final GenericContainer<?> couchdbContainer = new GenericContainer<>(DockerImageName.parse("couchdb:3.3.3"))
+    .withExposedPorts(5984)
+    .withEnv("COUCHDB_USER", "admin")
+    .withEnv("COUCHDB_PASSWORD", "password");
+
+  private static CouchdbClient client;
+  private static Vertx vertx;
+
+  @BeforeAll
+  static void setup() {
+    vertx = Vertx.vertx();
```

Comment:
There's a JUnit Vert.x extension that allows to inject vertx:
https://vertx.io/docs/vertx-junit5/java/#_integration_with_junit_5